app.py

```diff
@@ -29,15 +29,31 @@ def init_app(self):
         self._current_dialog = None
         self._current_panel = None
 
-        # Register the app as a panel.
-        self._unique_panel_id = self.engine.register_panel(self.create_panel)
-
-        # Register a menu entry on the ShotGrid menu so that users can launch the panel.
-        self.engine.register_command(
-            "Scene Breakdown...",
-            self.create_panel,
-            {"type": "panel", "short_name": "breakdown"},
-        )
+        if self.get_setting("app_display_mode") == "panel":
+
+            # Register the app as a panel.
+            self._unique_panel_id = self.engine.register_panel(self.create_panel)
+
+            # Register a menu entry on the ShotGrid menu so that users can launch the panel.
+            self.engine.register_command(
+                "Scene Breakdown...",
+                self.create_panel,
+                {"type": "panel", "short_name": "breakdown"},
+            )
+
+        elif self.get_setting("app_display_mode") == "dialog":
+
+            # Register the app as a dialog
+            # Register a menu entry on the ShotGrid menu so that users can launch the dialog.
+            self.engine.register_command(
+                "Scene Breakdown...",
+                lambda: self.create_dialog(),
+                {"type": "dialog", "short_name": "breakdown"},
+            )
+
+        else:
+            self.logger.error("An invalid app_display_mode was configured. `{}` is not a valid app_display_mode "
```

Comment:
Hello @RicardoMusch ,

The CI build pipeline fails because of this two line combination so I have added this snippet from the `black` hook linter:

```
self.logger.error(
                "An invalid app_display_mode was configured. `{}` is not a valid app_display_mode "
                "setting!".format(self.get_setting("app_display_mode"))
            )
```
I could not push this to your forked branch so just wanted to paste it here - thanks!